### PR TITLE
Csv inputter

### PR DESCRIPTION
**Pull request by DeemoONeill**

### CODE REVIEW

File: src/inputter.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    rows = [row for row in reader]
+
+
+# structure the data here
+split_recipe_name = recipe_name.split(",")
+cleaned_recipe_name = split_recipe_name[0]
+
+ingredients = []
+for row in rows:
+    ingredients.extend(row)
```
Comment (DeemoONeill) [inline src/inputter.py]: I don't know why you are flattening this here.

in the row you have a two item list

[weight and unit, ingredient]
so take the 0th element and clean it up

here's a hint, iterate over the string, and filter out any non-alphabetic character. if you do this twice, one to keep only the alphabetic, one to get rid of them, you have the unit and the amount

File: src/inputter.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ingredient_names = ingredients[1 : len(ingredients) : 2]
+
+
+weights_and_measures = ingredients[0 : len(ingredients) : 2]
```
Comment (DeemoONeill) [inline src/inputter.py]: you only have to do this because you've flattened everything

File: src/inputter.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+new_weights_and_measures = " ".join(weights_and_measures)
+split_weights_and_measures = new_weights_and_measures.split(" ")
+weights = split_weights_and_measures[0 : len(split_weights_and_measures) : 2]
+
+
+recipe = {cleaned_recipe_name: {"ingredients": ingredient_names}}
```
Comment (DeemoONeill) [inline src/inputter.py]: the perfect is the enemy of the good here.

you want a dictionary that looks like

{recipe name: {ingredients: {ingredient_name : [units]}

get the ingredient name and units into that format, put them in the dictionary and you're most of the way there, then you just have to clean up the units

**Comment by Rustyoneill** [comment]
i have pushed an update onto this, not sure if you can see it above this?

**Comment by Rustyoneill** [comment]
added new push

File: src/inputter.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    measure = ""
+    for char in value:
+        if char.isalpha():
+            unit += char
+        elif not char.isalpha():
+            measure += char
+    if unit == "":
+        unit = "unit"
+    if unit == "toserve":
+        break
```
Comment (DeemoONeill) [inline src/inputter.py]: continue instead of break here

File: src/inputter.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    if key == "":
+        break
```
Comment (DeemoONeill) [inline src/inputter.py]: same here

File: src/inputter.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    print(unit)
+    ingredients[key] = [measure, unit]
```
Comment (DeemoONeill) [inline src/inputter.py]: float(measure)

**Comment by Rustyoneill** [comment]
sure why not

**Review by Rustyoneill — APPROVED** [review]
sure

**Pull request merged**